src/Compilers/Core/Portable/Compilation/CompilationOptions.cs

```diff
@@ -361,6 +361,14 @@ public CompilationOptions WithReportSuppressedDiagnostics(bool value)
         }
 
         /// <summary>
+        /// Creates a new options instance with the deterministic property set accordingly.
+        /// </summary>
+        public CompilationOptions WithDeterministic(bool deterministic)
```

Comment:
@jaredpar No API change error (or warning in VS).  

You were right about the "new" shadow.  Vs did catch that one.  I think I ran my build incorrectly the first time through, but I caught it and changed it when I went back and looked.